src/main/java/org/opensearch/search/relevance/transformer/personalizeintelligentranking/reranker/impl/AmazonPersonalizedRankerImpl.java

```diff
@@ -66,12 +65,21 @@ public SearchHits rerank(SearchHits hits, PersonalizeRequestParameters requestPa
                         .collect(Collectors.toList());
             }
             logger.info("Document Ids to re-rank with Personalize: {}", Arrays.toString(documentIdsToRank.toArray()));
-            // TODO: Parse context from request parameters
             String userId = requestParameters.getUserId();
+            Map<String, String> context = requestParameters.getContext() != null ?
+                                            requestParameters.getContext().entrySet().stream()
+                                                    .filter(e -> e.getValue() instanceof String)
```

Comment:
Agree. Personalize API expects Map<String, String> for context input. So decided to throw exception instead of assuming toString() returns correct value.